tests/integration/test_charm.py

```diff
@@ -71,30 +33,37 @@ async def test_prometheus_scrape_relation_with_prometheus_tester(
     prometheus_app_name = "prometheus"
     tester_app_name = "prometheus-tester"
 
-    await ops_test.model.deploy(
-        prometheus_charm, resources=prometheus_resources, application_name=prometheus_app_name
-    )
-    await ops_test.model.deploy(
-        prometheus_tester_charm, resources=tester_resources, application_name=tester_app_name
-    )
-    await ops_test.model.wait_for_idle(apps=[prometheus_app_name], status="active")
-    await ops_test.model.wait_for_idle(apps=[tester_app_name], status="active")
-    await ops_test.model.block_until(
-        lambda: len(ops_test.model.applications[prometheus_app_name].units) > 0
+    app_names = [prometheus_app_name, tester_app_name]
+
+    await asyncio.gather(
+        ops_test.model.deploy(
+            prometheus_charm, resources=prometheus_resources, application_name=prometheus_app_name
+        ),
+        ops_test.model.deploy(
+            prometheus_tester_charm, resources=tester_resources, application_name=tester_app_name
+        ),
     )
+
+    await ops_test.model.wait_for_idle(apps=app_names, status="active")
```

Comment:
Could be moved into the gather?

Reply:
That would make it less clear imo, maybe even flaky, given that the `wait_for_idle` call then will (likely) be executed prior to there being any app entries to await.